### DISCUSSION TITLE
How to sort messages by their timestamp instead of offset

### DISCUSSION THREAD

**Discussion by risoms** [confluentinc/confluent-kafka-python#1762] [Q&A]
Hello all. I'm attempting to write a script that will consume the most recent message in a given topic. This is part of a integration test, which will attempt to verify that a given message was produced, and verify that the same message was consumed.

I'm noticing that the offsets are not sequential, which is impacting the order in which messages are received.

Is there any way using this library to poll only for a specific message?

**Comment by pranavrth** [comment]
You can use `seek` method present in `Consumer` to reach a particular offset and poll that message.
